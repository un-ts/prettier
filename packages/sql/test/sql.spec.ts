import prettier from 'prettier'

import SqlPlugin, { type SqlFormatOptions } from 'prettier-plugin-sql'

test('node-sql-parser', () => {
  const options: SqlFormatOptions = {
    filepath: 'test.sql',
    parser: 'sql',
    plugins: [SqlPlugin],
    pluginSearchDirs: false,
    formatter: 'node-sql-parser',
  }
  expect(prettier.format('SELECT * FROM `t`', options)).toMatchSnapshot()

  expect(
    prettier.format(
      'UPDATE a SET id = 1 WHERE name IN (SELECT name FROM b)',
      options,
    ),
  ).toMatchSnapshot()

  expect(
    prettier.format(
      'SELECT * FROM `t`;\nUPDATE a SET id = 1 WHERE name IN (SELECT name FROM b)',
      options,
    ),
  ).toMatchSnapshot()
})
