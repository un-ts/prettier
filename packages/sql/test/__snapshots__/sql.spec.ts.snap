// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`node-sql-parser 1`] = `
"SELECT * FROM \`t\`
"
`;

exports[`node-sql-parser 2`] = `
"UPDATE \`a\` SET \`id\` = 1 WHERE \`name\` IN (SELECT \`name\` FROM \`b\`)
"
`;

exports[`node-sql-parser 3`] = `
"SELECT * FROM \`t\` ; UPDATE \`a\` SET \`id\` = 1 WHERE \`name\` IN (SELECT \`name\` FROM \`b\`)
"
`;
